server/application-server/src/main/java/de/tum/in/www1/hephaestus/leaderboard/LeaderboardService.java

```diff
@@ -43,16 +42,12 @@ public LeaderboardService(
     }
 
     @Transactional
-    public List<LeaderboardEntryDTO> createLeaderboard(Optional<LocalDate> after, Optional<LocalDate> before,
+    public List<LeaderboardEntryDTO> createLeaderboard(Optional<OffsetDateTime> after, Optional<OffsetDateTime> before,
             Optional<String> repository) {
         logger.info("Creating leaderboard dataset");
-
-        LocalDateTime afterCutOff = after.isPresent() ? after.get().atStartOfDay()
-                : LocalDate.now().minusDays(timeframe).atStartOfDay();
-        Optional<LocalDateTime> beforeCutOff = before.map(date -> date.plusDays(1).atStartOfDay());
-
-        var afterOffset = afterCutOff.atOffset(ZoneOffset.UTC);
-        var beforeOffset = beforeCutOff.map(b -> b.atOffset(ZoneOffset.UTC)).orElse(OffsetDateTime.now());
+        var afterOffset = after.orElse(LocalDate.now().minusDays(timeframe).atStartOfDay().atOffset(ZoneOffset.UTC));
```

Comment:
Timeframe is also just for monitoring, the timeframe to sync. I think something is incorrect here.

IMO we are missing the following config:
 - `hephaestus.leaderboard.weekly.resetDay`, e.g. `2` for Tuesday
 - `hephaestus.leaderboard.weekly.resetTime`, e.g. `9:00` for 9 A.M.

Reply:
Since we now removed the `Optional`-part, this will be handled by the client.
So this should be obsolete now anyways :)